stream-transformation-tool-service/src/main/java/uk/gov/justice/tools/eventsourcing/transformation/service/EventStreamTransformationService.java

```diff
@@ -62,34 +72,49 @@ public void register(@Observes final EventTransformationFoundEvent event) throws
     public UUID transformEventStream(final UUID streamId) throws EventStreamException {
         final Stream<JsonEnvelope> eventStream = eventSource.getStreamById(streamId).read();
 
-        if (requiresTransformation(eventStream)) {
-            try {
-                final UUID clonedStreamId = eventSource.cloneStream(streamId);
+        switch (requiresTransformation(eventStream, streamId)) {
+            case TRANSFORM_EVENT:
+                return transformEvent(streamId, eventStream);
+            case NO_ACTION:
+                return null;
+            case ARCHIVE:
+                return archiveStream(streamId, eventStream);
+        }
+        return null;
+    }
 
-                if (logger.isDebugEnabled()) {
-                    logger.debug(format("Cloned stream '%s' from stream '%s'", clonedStreamId, streamId));
-                }
 
-                final EventStream stream = eventSource.getStreamById(streamId);
-                final Stream<JsonEnvelope> events = stream.read();
+    private UUID transformEvent(UUID streamId, Stream<JsonEnvelope> eventStream) {
+        try {
+            final UUID clonedStreamId = eventSource.cloneStream(streamId);
 
-                eventSource.clearStream(streamId);
+            if (logger.isDebugEnabled()) {
+                logger.debug(format("Cloned stream '%s' from stream '%s'", clonedStreamId, streamId));
+            }
 
-                logger.info("transforming events on stream {}", streamId);
-                final Stream<JsonEnvelope> transformedEventStream = transform(events);
+            final EventStream stream = eventSource.getStreamById(streamId);
+            final Stream<JsonEnvelope> events = stream.read();
 
-                stream.append(transformedEventStream.map(this::clearEventVersion));
-                events.close();
-            } catch (Exception e) {
-                logger.error("Failed to clone stream", e);
-            }
-            eventStream.close();
-            return streamId;
-        } else {
-            logger.debug("Stream {} did not require transformation", streamId);
-            eventStream.close();
-            return null;
+            eventSource.clearStream(streamId);
+
+            logger.info("transforming events on stream {}", streamId);
+            final Stream<JsonEnvelope> transformedEventStream = transform(events);
+
+            stream.append(transformedEventStream.map(this::clearEventVersion));
+            events.close();
+        } catch (Exception e) {
+            logger.error("Failed to clone stream", e);
```

Comment:
we need to say which stream id failed to clone

Reply:
changed the logger statement to accept the stream Id parameter.
Not quite sure about the exception as this is an existing implementation and not changed as part of this work.